plugins/connection/httpapi.py

```diff
@@ -169,37 +179,41 @@ def __init__(self, play_context, new_stdin, *args, **kwargs):
         )
 
         self._auth = None
-
         if self._network_os:
+          self.load_platform_plugins(self._network_os)
```

Comment:
How does this get called if network_os is not set?

Reply:
It can be invoked from either action plugin or module code. Please refer the linked PR for cisco.ios collection